hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3087,37 +3121,48 @@ public S3VolumeContext getS3VolumeContext() throws IOException {
     String userPrincipal = Server.getRemoteUser().getShortUserName();
 
     if (s3Auth != null) {
-      String accessID = s3Auth.getAccessId();
-      // TODO HDDS-6063: Volume lock is needed here along with the other
-      //  multi-tenant read requests.
+      String accessId = s3Auth.getAccessId();
       Optional<String> optionalTenantId =
-          multiTenantManager.getTenantForAccessID(accessID);
+          multiTenantManager.getTenantForAccessID(accessId);
 
       if (optionalTenantId.isPresent()) {
-        String tenantId = optionalTenantId.get();
+        final String tenantId = optionalTenantId.get();
+
         OmDBTenantInfo tenantInfo =
             metadataManager.getTenantStateTable().get(tenantId);
         if (tenantInfo != null) {
           s3Volume = metadataManager.getTenantStateTable().get(tenantId)
               .getBucketNamespaceName();
         } else {
           String message = "Expected to find a tenant for access ID " +
-              accessID +
+              accessId +
               " but no tenant was found. Possibly inconsistent OM DB!";
           LOG.error(message);
           throw new OMException(message, ResultCodes.TENANT_NOT_FOUND);
```

Comment:
The tenant may have been removed since we are not holding a lock. We can lighten up the message and log it as a warning instead of an error. We should keep the exception though.

Reply:
done